enforcer-rules/src/main/java/io/jenkins/tools/incrementals/enforcer/RequireExtensionVersion.java

```diff
@@ -25,32 +25,60 @@
 package io.jenkins.tools.incrementals.enforcer;
 
 import java.util.List;
+import java.util.Objects;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
+import javax.inject.Inject;
+import javax.inject.Named;
 import org.apache.maven.AbstractMavenLifecycleParticipant;
+import org.apache.maven.artifact.versioning.ArtifactVersion;
 import org.apache.maven.artifact.versioning.DefaultArtifactVersion;
+import org.apache.maven.artifact.versioning.InvalidVersionSpecificationException;
+import org.apache.maven.artifact.versioning.VersionRange;
+import org.apache.maven.enforcer.rule.api.AbstractEnforcerRule;
 import org.apache.maven.enforcer.rule.api.EnforcerRuleException;
-import org.apache.maven.enforcer.rule.api.EnforcerRuleHelper;
-import org.apache.maven.plugin.logging.Log;
-import org.apache.maven.plugins.enforcer.AbstractVersionEnforcer;
+import org.codehaus.plexus.PlexusContainer;
 import org.codehaus.plexus.classworlds.realm.ClassRealm;
 import org.codehaus.plexus.component.repository.exception.ComponentLookupException;
+import org.codehaus.plexus.util.StringUtils;
 
 /**
  * Verifies that {@code git-changelist-maven-extension}, if present, is sufficiently new.
  */
-public class RequireExtensionVersion extends AbstractVersionEnforcer {
+@Named("requireExtensionVersion")
+public class RequireExtensionVersion extends AbstractEnforcerRule {
 
     private static final Pattern ID_PATTERN = Pattern.compile("\\QcoreExtension>io.jenkins.tools.incrementals:git-changelist-maven-extension:\\E(.+)");
 
+    /**
+     * Specify the required version. Some examples are:
+     * <ul>
+     * <li><code>2.0.4</code> Version 2.0.4 and higher (different from Maven meaning)</li>
+     * <li><code>[2.0,2.1)</code> Versions 2.0 (included) to 2.1 (not included)</li>
+     * <li><code>[2.0,2.1]</code> Versions 2.0 to 2.1 (both included)</li>
+     * <li><code>[2.0.5,)</code> Versions 2.0.5 and higher</li>
+     * <li><code>(,2.0.5],[2.1.1,)</code> Versions up to 2.0.5 (included) and 2.1.1 or higher</li>
+     * </ul>
+     *
+     * @see {@link #setVersion(String)}
+     * @see {@link #getVersion()}
+     */
+    private String version;
+
+    private final PlexusContainer container;
+
+    @Inject
+    public RequireExtensionVersion(PlexusContainer container) {
+        this.container = Objects.requireNonNull(container);
+    }
```

Comment:
Adapting to the new recommended way of injecting components via constructor (`EnforcerRuleHelper` is now deprecated).